src/index.ts

```diff
@@ -0,0 +1,236 @@
+import createManipulablePromise, {
+  ManipulablePromise
+} from "./utils/manipulable-promise";
+import {
+  IncomingMessage,
+  OutgoingMessage,
+  JSONifiedMessage,
+  Stringifyable
+} from "./message";
+import { TALKER_TYPE, TALKER_ERR_TIMEOUT } from "./strings";
+
+interface SentMessages {
+  [id: number]: ManipulablePromise<IncomingMessage | Error>;
+}
+
+/**
+ * Talker
+ * Opens a communication line between this window and a remote window via postMessage.
+ */
+class Talker {
+  /*
+   * @property timeout - The number of milliseconds to wait before assuming no response will be received.
+   */
+  public timeout: number = 3000;
```

Comment:
Nit: it's always helpful to have the units as part of the variable name:

```suggestion
  public timeoutMs: number = 3000;
```

Reply:
I want to avoid bumping the major version number and this would be changing the public API.